python/cudf/cudf/core/column/column.py

```diff
@@ -120,11 +120,11 @@ def mask_array_view(self):
     def __len__(self):
         return self.size
 
-    def to_pandas(self, index=None, nullable_pd_dtype=False, **kwargs):
+    def to_pandas(self, index=None, **kwargs):
         pd_series = self.to_arrow().to_pandas(**kwargs)
         if index is not None:
             pd_series.index = index
-        return pd_series
+        return pd_series.copy()
```

Comment:
If arrow is taking the stance that giving someone an immutable Pandas dataframe is okay, any reason we shouldn't follow suite?

Reply:
I think returning something immutable would lead to headaches more likely than not. The 'contract' of `to_pandas()` is to give back the user a pandas object representing the cuDF data. If we can modify cuDF dataframes, and in general modify pandas dataframes that are constructed through other means, it would be a little strange to produce an immutable dataframe here just because we went through arrow to get there. IMO the usefulness of arrow here is mostly being able to punt on the logic around nulls.

Reply:
Maybe we should raise an Arrow JIRA to see if there would be interested in adding a kwarg or something similar to enable mutability on the Pandas side. Even if we use the experimental `self_destruct=True` option which blows up the PyArrow object, it still leaves the Pandas object as read-only which is a bit surprising.

Alternatively, maybe we could ask if there's a workaround given we know the Arrow object is only a temporary specifically to get to Pandas in this case.

Reply:
This just looks very expensive to do a full deep copy here.

Reply:
cc @jorisvandenbossche if you have any ideas for workarounds here

Reply:
According to the arrow docs it looks like arrow is actually going to make a copy of the underlying data in the majority of the situations already - really any time we're not dealing with numeric columns that have `null_count==0`. Consequently we don't get the read only issue in those scenarios, even if there's only a single null, the final pandas dataframe ends up being mutable.

One thing we should definitely do is not _re-copy_ those columns. This is something we can detect and avoid at the column level which would be pretty easy.

That still leaves the issue of the columns that arrow _does_ zero copy though. It does seem like there should be some way of releasing the memory here...

Reply:
Given the only place this is happening is for a Column of numeric data without nulls, maybe we just go through numpy for now in that situation? Should still be zero copy for Pandas:
```
import numpy as np
import pandas as pd

a = np.array([1, 2, 3])
print(a.__array_interface__['data'][0])

b = pd.Series(a)
print(b.values.__array_interface__['data'][0])
```

Reply:
That makes a lot of sense 👍

Reply:
Please open a JIRA about this or bring it up on the mailing list. As I think the use case being described here certainly makes sense, and is something pyarrow should enable (specifically the fact that with `self_destruct=True` you still get a read-only seems strange to me, although there might be reasons I am not aware of)

Reply:
Thanks Joris! @brandon-b-miller can you please open a JIRA about this?